pkg/distro/defs/id_test.go

```diff
@@ -0,0 +1,46 @@
+package defs
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMatchAndNormalizeHappy(t *testing.T) {
+	for _, tc := range []struct {
+		reStr, nameVer string
+		expected       string
+	}{
+		// simple cases, no capture groups
+		{`rhel-10\.[0-9]{1,2}`, "rhel-100", ""},
+		{`rhel-10\.[0-9]{1,2}`, "rhel-10.0", "rhel-10.0"},
+		// capture groups for major/minor
+		{`(?P<name>rhel)-(?P<major>8)\.?(?P<minor>[0-9]{1,2})`, "rhel-8.10", "rhel-8.10"},
+		{`(?P<name>rhel)-(?P<major>8)\.?(?P<minor>[0-9]{1,2})`, "rhel-810", "rhel-8.10"},
+		// capture groups for just major
+		{`(?P<name>centos)-(?P<major>[0-9])stream`, "centos-9stream", "centos-9"},
+		// normalizing strange things works
+		{`(?P<major>[0-9])-(?P<name>foo)`, "8-foo", "foo-8"},
+	} {
+		found, err := matchAndNormalize(tc.reStr, tc.nameVer)
+		assert.NoError(t, err)
+		assert.Equal(t, found, tc.expected)
+	}
+}
+
+func TestMatchAndNormalizeSad(t *testing.T) {
+	for _, tc := range []struct {
```

Comment:
I would suggest to also test a case when the regex has a valid number of named match groups, but the names of the match groups are not what the code expects, i.e. `(?P<blabla1>rhel)-(?P<blabla2>8)\.?(?P<blabla3>[0-9]{1,2})`.

Reply:
I had the same thought